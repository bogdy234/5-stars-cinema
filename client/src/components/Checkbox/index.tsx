import { FC, ReactElement } from "react";

interface CheckboxProps {
  checked: boolean;
  onChange: () => void;
  label: string;
}

const Checkbox: FC<CheckboxProps> = ({
  checked,
  onChange,
  label,
}): ReactElement => {
  return (
    <div className="flex items-center gap-10">
      <div className="text-xl w-40">{label}</div>
      <input type="checkbox" checked={checked} onChange={onChange} />
    </div>
  );
};

export default Checkbox;
