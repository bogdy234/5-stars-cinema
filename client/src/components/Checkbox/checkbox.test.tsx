import renderer from "react-test-renderer";
import Checkbox from "./index";

it("renders correctly", () => {
  const tree = renderer
    .create(<Checkbox checked={false} onChange={jest.fn()} label={""} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
