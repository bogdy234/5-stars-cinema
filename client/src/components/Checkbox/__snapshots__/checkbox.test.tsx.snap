// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex items-center gap-10"
>
  <div
    className="text-xl w-40"
  >
    
  </div>
  <input
    checked={false}
    onChange={[MockFunction]}
    type="checkbox"
  />
</div>
`;
